### DISCUSSION TITLE
Html document parser

### DISCUSSION THREAD

**Discussion by TeddyAlbina** [RehanSaeed/Schema.NET#142] [Q&A]
### Question

Hello, i want to know if this library can be used to extract microdata from an html document ? All the samples in the front page are about generating microdata.

Thanks.

**Comment by Turnerj** [comment]
Currently, while Schema.NET can represent the markup from Microdata, the library itself can't parse it from HTML. What Schema.NET can do is parse and generate JSON-LD.

Functionality to allow the parsing of Microdata and RDFa might be a beneficial extension at a point of time in the future.

**Comment by RehanSaeed** [comment]
You can use the `HtmlAgilityPack` to parse the JSON-LD from the HTML. I'd take a PR to add a sample or some docs showing how to do this.
